tests/dates/test_BaseDatetimeTransformer.py

```diff
@@ -34,32 +41,43 @@ def test_non_datetypes_error(
         bad_value,
         bad_type,
     ):
-        "Test that transform raises an error if columns contains non date types"
+        "Test that transform raises an error if columns contains non datetime types"
 
         args = minimal_attribute_dict[self.transformer_name].copy()
-
-        # pull out columns arg to use below, and remove from dict
         columns = args["columns"]
 
+        transformer = uninitialized_transformers[self.transformer_name](
+            **args,
+        )
+
+        df = deepcopy(minimal_dataframe_lookup[self.transformer_name])
+
+        # if transformer is not yet polars compatible, skip this test
+        if not transformer.polars_compatible and isinstance(df, pl.DataFrame):
+            return
+
         for i in range(len(columns)):
-            df = deepcopy(minimal_dataframe_lookup[self.transformer_name])
             col = columns[i]
-            df[col] = bad_value
-            # force date test to wanted type
-            if bad_type == "Date":
-                df[col] = df[col].astype("date32[pyarrow]")
-
-            x = uninitialized_transformers[self.transformer_name](
-                **args,
-            )
+            # force date test to wanted type if pandas df, to avoid
```

Comment:
S: Worth including a similar comment into the `dates.py` base class? It took me a while to understand why `check_columns_are_date_or_datetime()` is structured the way that it is without this comment ... which is in the tests.

Reply:
I think that `check_columns_are_date_or_datetime` in general could do with some deep thinking, and that should be done when considering the other date transformers in the `dates.py` file. Won't hold on this item.